runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -10363,82 +10340,66 @@ genHeapAlloc(TR::Node * node, TR::Instruction *& iCursor, bool isVariableLen, TR
                       generateS390MemoryReference(metaReg, offsetof(J9VMThread, heapAlloc), cg), iCursor);
-      TR::LabelSymbol * fillerRemLabel = generateLabelSymbol(cg);
-
-      TR::LabelSymbol * fillerLoopLabel = generateLabelSymbol(cg);
 
+      // do this clear, if disableBatchClear is on
       if (inlineZeroInitialization)
          {
-         int32_t loops = allocSize / 256;
-         // If number of loops is 2 or less, consequent XC instructions is prefered over BRTC loops.
-         bool generateXCLoop = loops > 2;
-         TR::Register * shiftReg = NULL;
-         TR::Register * addressReg = resReg;
-         if(isVariableLen || generateXCLoop)
-            {
-            shiftReg = cg->allocateRegister();
-            addressReg = cg->allocateRegister();
-            if (conditions != NULL)
-               {
-               conditions->resetIsUsed();
-               conditions->addPostCondition(shiftReg, TR::RealRegister::AssignAny);
-               conditions->addPostCondition(addressReg, TR::RealRegister::AssignAny);
-               }
-
-            iCursor = generateRRInstruction(cg, TR::InstOpCode::getLoadRegOpCode(), node, addressReg, resReg, iCursor);
-            }
-
          if(isVariableLen)
             {
+            // Repurpose sizeReg to keep loop count.
+            TR::Register * loopCountRegister = sizeReg;
+
             if (TR::comp()->target().is64Bit())
                {
-               iCursor = generateRSInstruction(cg, TR::InstOpCode::SRAG, node, shiftReg, lengthReg, 8, iCursor);
+               iCursor = generateRSInstruction(cg, TR::InstOpCode::SRAG, node, loopCountRegister, sizePreserveRegister, 8, iCursor);
                }
             else
                {
                if (comp->target().cpu.isAtLeast(OMR_PROCESSOR_S390_Z196))
                   {
-                  iCursor = generateRSInstruction(cg, TR::InstOpCode::SRAK, node, shiftReg, lengthReg, 8, iCursor);
+                  iCursor = generateRSInstruction(cg, TR::InstOpCode::SRAK, node, loopCountRegister, sizePreserveRegister, 8, iCursor);
                   }
                else
                   {
-                  iCursor = generateRRInstruction(cg, TR::InstOpCode::LR, node, shiftReg, lengthReg, iCursor);
-                  iCursor = generateRSInstruction(cg, TR::InstOpCode::SRA, node, shiftReg, 8, iCursor);
+                  iCursor = generateRRInstruction(cg, TR::InstOpCode::LR, node, loopCountRegister, sizePreserveRegister, iCursor);
+                  iCursor = generateRSInstruction(cg, TR::InstOpCode::SRA, node, loopCountRegister, 8, iCursor);
                   }
                }
 
+            TR::Register * addressReg = srm->findOrCreateScratchRegister();
+            iCursor = generateRRInstruction(cg, TR::InstOpCode::getLoadRegOpCode(), node, addressReg, resReg, iCursor);
+
+            TR::LabelSymbol * ResidueBytesZeroingLabel = generateLabelSymbol(cg);
```

Comment:
variable names are lowerCamelCase throughout the code-base. Please don't diverge from that.